widget/embedded/src/store/slices/data.ts

```diff
@@ -346,4 +324,50 @@ export const createDataSlice: StateCreator<
       throw error;
     }
   },
+  setSupportedTokens: (options) => {
+    const tokens = get()._tokensMapByBlockchainName;
+    let nextState: Record<TokenHash, Token> = {};
+    if (Array.isArray(options.data.tokens)) {
+      options.data.tokens.forEach((token) => {
+        const tokenHash = createTokenHash(token);
+        const tokenFromMeta = tokens?.[token.blockchain]?.[tokenHash];
+        if (tokenFromMeta) {
+          nextState[tokenHash] = tokenFromMeta;
+        }
+      });
+    } else {
+      if (!options.data.blockchains) {
+        nextState = Object.fromEntries(get()._tokensMapByTokenHash.entries());
+      }
+      options.data.blockchains?.forEach((blockchain) => {
+        const value = !Array.isArray(options.data.tokens)
+          ? options.data.tokens?.[blockchain]
+          : undefined;
+        if ((!value || value?.isExcluded) && tokens?.[blockchain]) {
```

Comment:
Excluded tokens do not work properly.
It seems that this condition is not correct.